Source/WebCore/platform/SharedBuffer.cpp

```diff
@@ -67,6 +74,38 @@ Ref<FragmentedSharedBuffer> FragmentedSharedBuffer::create(DataSegment::Provider
     return adoptRef(*new FragmentedSharedBuffer(WTFMove(provider)));
 }
 
+#if PLATFORM(COCOA)
+std::optional<Ref<FragmentedSharedBuffer>> FragmentedSharedBuffer::fromIPCData(IPCData&& ipcData)
+{
+    return WTF::switchOn(WTFMove(ipcData), [](Vector<std::span<const uint8_t>>&& data) -> std::optional<Ref<FragmentedSharedBuffer>> {
+        if (!data.size())
+            return SharedBuffer::create();
+
+        CheckedSize size = 0;
+        for (auto span : data)
+            size += span.size();
+
```

Comment:
I think we want to deal with overflow here and early return:
```
if (size.hasOverflowed())
    return std::nullopt;
```

Otherwise, we'll crash when using the `size` below.